conf/config-schema.json

```diff
@@ -0,0 +1,16 @@
+{
+    "type": "object",
+    "properties": {
+        "root": { "type": "boolean" },
+        "globals": { "type": ["object", "array"] },
+        "global": { "type": ["object", "array"] },
+        "parser": { "type": ["string", "null"] },
+        "env": { "type": "object" },
+        "plugins": { "type": ["array"] },
+        "settings": { "type": "object" },
+        "extends": { "type": ["string", "array"] },
+        "rules": { "type": "object" },
+        "parserOptions": { "type": "object" }
+    },
+    "additionalProperties": false
```

Comment:
This is already a "breaking" PR and should be semver-major-- I think we could safely error for ecmaFeatures as long as we include something in the 4.0.0 migration guide. (Just my two cents.)